packages/gatsby/src/utils/start-server.ts

```diff
@@ -705,7 +705,7 @@ module.exports = {
           res.send(e).status(500)
         }
       } else {
-        const potentialPagePath = reverseFixedPagePath(req.path)
+        const potentialPagePath = reverseFixedPagePath(decodeURI(req.path))
```

Comment:
not fully sure about this one, but this particular one was causing issues with https://github.com/gatsbyjs/gatsby/blob/67c82e2b862fe39ca6af6d4060ace43c55f7f20d/e2e-tests/development-runtime/cypress/integration/navigation/linking.js#L113 (browsers under the hood will encode those).

Other option is to try `req.path` as is and decoded one separately? but given that this seems to make tests happy in general it should be fine?

Reply:
I wonder if this is something cloud needs to fix too